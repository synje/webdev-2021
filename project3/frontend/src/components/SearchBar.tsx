import React from 'react'
import { Box, Button, FormControl, FormLabel, Input } from "@chakra-ui/react"


/**
 * Exports SearchBar component to component DBList.tsx
 * Searchbar input-field and search-button
 */
export default class SearchBar extends React.Component {
    props: any


    render() {
        return (
            <>
                <Box display="flex" justifyContent="center">
                    <Box className="searchBar" display="flex" justifyContent="center" w="100%">
                        <FormControl display="flex" justifyContent="center" w="90%">
                            <FormLabel htmlFor="searchInput" display="none">Search</FormLabel>
                            <Input type="text"
                                data-cy="searchBar"
                                className="searchBar"
                                name="searchInput"
                                placeholder={this.props.placeholder}
                                onChange={this.props.searchInput}
                            />
                            <Button type="submit"
                                data-cy="searchButton"
                                color="darkslategray"
                                className="submitbtn"
                                bg="floralwhite"
                                border="0.2px solid dimgray"
                                onClick={this.props.handleClick}
                            >SEARCH</Button>
                        </FormControl>
                    </Box>
                </Box>
            </>
        )
    }
}