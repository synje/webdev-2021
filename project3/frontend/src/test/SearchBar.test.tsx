import { fireEvent } from '@testing-library/dom';
import { render } from '@testing-library/react';
import { useState } from 'react';
import renderer from 'react-test-renderer'
import SearchBar from "../components/SearchBar";

test('Snapshot SearchBar', () => {
    const tree = renderer.create(<SearchBar />)
    
    expect(tree).toMatchSnapshot()

})

test('Test input from searchbar', () =>{
    let searchInputField = ""
    const { getByPlaceholderText }= render(
            <SearchBar 
            placeholder="Search..." 
            searchInput={(e: any) => searchInputField = e.target.value}
            />
    )

    // const inputField = component.root.findByType('input')
    const inputField = getByPlaceholderText('Search...')
    expect(searchInputField).toBe('')
    fireEvent.change(inputField,{target: {value: "Avril"}})
    expect(searchInputField).toBe('Avril')


})

test('Test button specified action works', () => {
    let clickWorks = false;
    const component = renderer.create(
            <SearchBar 
            handleClick={(e: any)=> {
                clickWorks = true
            }}
            />
    )
    expect(clickWorks).toBe(false)
    const button = component.root.findByType('button')
    button.props.onClick()
    expect(clickWorks).toBe(true)
})