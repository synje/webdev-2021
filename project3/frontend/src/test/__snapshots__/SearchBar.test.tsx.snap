// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot SearchBar 1`] = `
<div
  className="css-1l4w6pd"
>
  <div
    className="searchBar css-ldo4d5"
  >
    <div
      className="chakra-form-control css-1z0y902"
      role="group"
    >
      <label
        className="chakra-form__label css-1v3sb5r"
        htmlFor="searchInput"
        id="field-3559519501-1-label"
      >
        Search
      </label>
      <input
        className="chakra-input searchBar css-0"
        data-cy="searchBar"
        disabled={false}
        id="field-3559519501-1"
        name="searchInput"
        onBlur={[Function]}
        onFocus={[Function]}
        readOnly={false}
        required={false}
        type="text"
      />
      <button
        className="chakra-button submitbtn css-1y2wz87"
        data-cy="searchButton"
        type="submit"
      >
        SEARCH
      </button>
    </div>
  </div>
</div>
`;
